### ISSUE TITLE
OpenType-SVG should respect currentColor

### ISSUE THREAD

**Issue by jkwchui** [typst/typst#4973]
### Description

This is for the dev version 0.11.1, which has OpenType-SVG support.

The SVG glyphs can contain fill or strokes with `currentColor` ([spec](https://learn.microsoft.com/en-us/typography/opentype/spec/svg#colors)).  These should receive `color` attributes from Typst markup; instead, it currently stays black and all the colors inside the SVG is impossible to change.

### Operating system

macOS

### Typst version

- [X] I am using the latest version of Typst

**Comment by Enivex** [comment]
Duplicate of https://github.com/typst/typst/issues/1939

**Comment by jkwchui** [comment]
I don't think this is a duplicate.  #1939 is about general rendering and changing colors in SVG, and involves inventing an API within Typst; it's a feature request.  This is a meeting a specification of a format that Typst (intends to) support, and could be as simple as pre-processing the SVG prior to passing into `usvg` with the text-color substituted into `currentColor`.

**Comment by Enivex** [comment]
> I don't think this is a duplicate. #1939 is about general rendering and changing colors in SVG, and involves inventing an API within Typst; it's a feature request. This is a meeting a specification of a format that Typst (intends to) support (hence a bug-fix), and could be as simple as pre-processing the SVG prior to passing into `usvg` with the text-color substituted into `currentColor`.

See the discussion in the two last replies

**Comment by LaurenzV** [comment]
Yes, this is a bit different.

**Comment by LaurenzV** [comment]
This is blocked by https://github.com/RazrFalcon/resvg/issues/768, though.

**Comment by LaurenzV** [comment]
@jkwchui Do you happen to know any fonts that use that feature, so I can try it? I haven't been able to find one...

**Comment by jkwchui** [comment]
I can send you one; let me try to find your email.

**Comment by LaurenzV** [comment]
All good, I've managed to make one myself now, but thanks! 😄

**Comment by ecstrema** [comment]
A workaround currently is direct string replace:

```typ
#let contents = read("file.svg")
#contents = contents.replace("currentColor", "red")

#image(bytes(contents))
```
